composer/trainer/checkpoint_hparams.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2021 MosaicML. All Rights Reserved.
+
+import logging
+import os
+import random
+import warnings
+from dataclasses import dataclass
+from typing import Any, Dict, Optional
+
+import numpy as np
+import torch
+import yahp as hp
+import yaml
+
+from composer.core.types import StateDict
+from composer.trainer.checkpoint import CheckpointLoader
+from composer.trainer.devices.device import Device
+
+log = logging.getLogger(__name__)
+
+
+@dataclass
+class CheckpointLoaderHparams(hp.Hparams):
+    """Hparams for the :class:`CheckpointLoader`.
+
+    See the documentation for the :class:`CheckpointLoader`.
+    """
+    checkpoint_filepath: Optional[str] = hp.optional(doc="Path to the serialized state_dict to recover state from.",
```

Comment:
I think this should be required. The `CheckpointLoaderHparams` in the `TrainerHparms` can be none.

Reply:
Agreed, changing this to be required.

Reply:
This has been addressed!